sdks/java/core/src/main/java/org/apache/beam/sdk/runners/PTransformReplacements.java

```diff
@@ -15,20 +15,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.runners.core.construction;
+package org.apache.beam.sdk.runners;
 
 import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
 
 import java.util.Map;
 import java.util.Set;
-import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Internal;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.values.PCollection;
-import org.apache.beam.sdk.values.PValue;
 import org.apache.beam.sdk.values.TupleTag;
 import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterables;
 
 /** */
+@Internal
```

Comment:
Why did this (and related classes) have to move?
I could be convinced otherwise to keep the move but it looks like long term we would want to get rid of Pipeline#replaceAll some day since runners should only doing proto -> proto conversions.

The only place I could find it in was PipelineTest. Can we instead move that test to somewhere in runners core construction?

Reply:
I don't agree with this. There are many reasons

 - Protos are about wire formats and often not preferred. One time they are very much not preferred is when the proto is a low-level representation of a higher-level concept, like a graph.
 - The SDKs are specifically designed to be the best API for writing correct Beam subgraphs. Using them is better than direct graph manipulation (and graph manipulation is better than proto manipulation).

I may have misunderstood what you meant by "proto -> proto" conversions. But if it is to serve the same purpose as `Pipeline#replaceAll` then I think it should look more or less the same. We could remove `Pipeline#replaceAll` from the core SDK and have an enhanced version just for runners, but that is a lot of complexity.

Reply:
We agree on replaceAll like functionality not existing in sdks core but I don't think having yet another layer on top of the protos would be much better then a few static methods that do common things that are necessary to easily walk and mutate a graph.